### PR TITLE
Fix More Info bug that reverted text to English

### PR DESCRIPTION
**Pull request by suecarmol**
[//]: # (Thank you for uploading a PR to the Wikipedia Library!)

## Description
Fixed a bug where the `More Info` button would revert back to English instead of a user's chosen language when the button was clicked again.

## Rationale
[//]: # (Why is this change required? What problem does it solve?)
Fixes a bug that was created a less enjoyable experience for non-English users.

## Phabricator Ticket
[//]: # (Link to the Phabricator ticket)
[T287577](https://phabricator.wikimedia.org/T287577)

## How Has This Been Tested?
[//]: # (- Did you add tests to your changes? Did you modify tests to accommodate your changes?)
[//]: # (- Can this change be tested manually? How?)
Tested manually in the interface

## Screenshots of your changes (if appropriate):
[//]: # (It can also be a GIF to prove that your changes are working)

**Before**
Before_fix

**After**
After_fix

## Types of changes
What types of changes does your code introduce? Add an `x` in all the boxes that apply:
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Minor change (fix a typo, add a translation tag, add section to README, etc.)

### CODE REVIEW

**Comment by Samwalton9** [comment]
From looking at the gifs am I right in thinking that the 'Less info' button still displays in English?

**Comment by suecarmol** [comment]
> From looking at the gifs am I right in thinking that the 'Less info' button still displays in English?

Yes, `Less Info` does not have a translation tag because it is originated from javascript (see [this line])(https://github.com/WikipediaLibrary/TWLight/blob/fa8be46a8d4cf38bfe3916b1fdd4884e7df9f504/TWLight/templates/homepage.html#L376)

**Comment by suecarmol** [comment]
We could replace `More info` and `Less info` with `Flip`.

**Comment by Samwalton9** [comment]
I think Flip is a lot less clear. Is there no way to mark text in JS as translatable?

**Comment by suecarmol** [comment]
> I think Flip is a lot less clear. Is there no way to mark text in JS as translatable?

Looks like we can! I passed the `Less info` string from the view so it can be translatable.

File: TWLight/templates/homepage.html [hunk 1]
```diff
@@ -370,9 +375,9 @@
       // If the invisible class exists in the partner image, then the button
       // needs to show "Less info"
       if (partnerImage.classList.contains("invisible")) {
-        partnerButton.innerHTML = "Less info";
+        partnerButton.innerHTML = lessInfo;
```
Comment (jsnshrmn) [inline TWLight/templates/homepage.html:378]:
```suggestion
        {% comment %}Translators: This text will show when a user clicks on the More Info button and the partner description shows.{% endcomment %}
        partnerButton.innerHTML = "{% trans 'Less info' %}";
```

File: TWLight/templates/homepage.html [hunk 2]
```diff
@@ -370,9 +375,9 @@
...
       } else {
-        partnerButton.innerHTML = "More info";
+        partnerButton.innerHTML = partnerText;
```
Comment (jsnshrmn) [inline TWLight/templates/homepage.html:380]:
```suggestion
        {% comment %}Translators: A button that reveals more information about a collection.{% endcomment %}
        partnerButton.innerHTML = "{% trans 'More info' %}";
```

File: TWLight/templates/homepage.html [hunk 3]
```diff
@@ -331,6 +331,12 @@
     $(function () {
       $('[data-toggle="tooltip"]').tooltip()
     });
+    // Getting the "More info" translation from the first button element
+    var partnerButtons = document.getElementsByClassName("partner-glider-button");
+    var partnerText = partnerButtons[0].innerText;
+    // Getting the Less info text from the view
+    var lessInfo = "{{ less_info }}";
+
```
Comment (jsnshrmn) [inline TWLight/templates/homepage.html:339]:
```suggestion

```

File: TWLight/views.py [hunk 4]
```diff
@@ -181,6 +181,9 @@ def get_context_data(self, **kwargs):
             )
         context["partners"] = partners_obj
 
+        # Translators: This text will show when a user clicks on the More Info button and the partner description shows.
+        context["less_info"] = _("Less info")
+
```
Comment (jsnshrmn) [inline TWLight/views.py:186]:
```suggestion

```

**Review by jsnshrmn — CHANGES_REQUESTED** [review]
We should be able to handle all of this in the template, since that's where the strings live.

**Comment by suecarmol** [comment]
Thanks for suggesting these changes! It makes the code more legible

**Review by jsnshrmn — APPROVED** [review]

**Comment by jsnshrmn** [comment]
thanks!

**Pull request merged**